plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php

```diff
@@ -250,10 +337,59 @@ protected function render( $attributes, $content, $block ) {
 			)
 		);
 
+		if ( $is_stepper_style ) {
+			$context    = array();
+			$product_id = $product->get_id();
+
+			$context['products'] = array();
+
+			if ( $product->is_type( 'grouped' ) ) {
+				// Add parent product.
+				$context['products'][ $product_id ] = array(
+					'id'                  => $product_id,
+					'type'                => 'grouped',
+					'quantity'            => 0,
+					'quantityConstraints' => $this->get_quantity_constraints( $product ),
+					'parentProductId'     => null,
+				);
+
+				// Add child products.
+				foreach ( $product->get_children() as $child_product_id ) {
+					$child_product = wc_get_product( $child_product_id );
+					if ( $child_product && $child_product->is_purchasable() && $child_product->is_in_stock() && ! $child_product->has_options() ) {
+						$context['products'][ $child_product_id ] = array(
+							'id'                  => $child_product_id,
+							'type'                => $child_product->get_type(),
+							'quantity'            => 0,
+							'quantityConstraints' => $this->get_quantity_constraints( $child_product, true ),
+							'parentProductId'     => $product_id,
+						);
+					}
+				}
+			} else {
+				$context['products'][ $product_id ] = array(
+					'id'                  => $product_id,
+					'type'                => $product->get_type(),
+					'quantity'            => 0,
+					'quantityConstraints' => $this->get_quantity_constraints( $product ),
+					'parentProductId'     => null,
+				);
```

Comment:
This code seems to be the same in the `if` and in the `else`? Could it be moved outside?